.NET build in pull request validation fails due to missing assemblies

Since at least 3 weeks, the pull request build validation seems to be broken due to missing assemblies:
https://github.com/ameyer505/D365FOAdminToolkit/actions/runs/11221423892/job/31191791301#step:7:377

>   D:\a\D365FOAdminToolkit\D365FOAdminToolkit\Project\D365FOAdminToolkit\D365FOAdminToolkitNET\AdminToolkitNET.cs(11,26): error CS0234: The type or namespace name 'Ax' does not exist in the namespace 'Microsoft.Dynamics' (are you missing an assembly reference?) [D:\a\D365FOAdminToolkit\D365FOAdminToolkit\Project\D365FOAdminToolkit\D365FOAdminToolkitNET\D365FOAdminToolkitNET.csproj]
  D:\a\D365FOAdminToolkit\D365FOAdminToolkit\Project\D365FOAdminToolkit\D365FOAdminToolkitNET\AdminToolkitNET.cs(12,38): error CS0234: The type or namespace name 'MetaModel' does not exist in the namespace 'Microsoft.Dynamics.AX.Metadata' (are you missing an assembly reference?) [D:\a\D365FOAdminToolkit\D365FOAdminToolkit\Project\D365FOAdminToolkit\D365FOAdminToolkitNET\D365FOAdminToolkitNET.csproj]
  D:\a\D365FOAdminToolkit\D365FOAdminToolkit\Project\D365FOAdminToolkit\D365FOAdminToolkitNET\AdminToolkitNET.cs(300,44): error CS0246: The type or namespace name 'AxQuerySimpleDataSource' could not be found (are you missing a using directive or an assembly reference?) [D:\a\D365FOAdminToolkit\D365FOAdminToolkit\Project\D365FOAdminToolkit\D365FOAdminToolkitNET\D365FOAdminToolkitNET.csproj]

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -80,6 +80,25 @@ jobs:
           version: ${{ matrix.version }}
           packagesDirectory: '${{ github.workspace }}/Project/D365FOAdminToolkit/packages'
 
+      - name: List FSC packages
+        run: ls -R Project/D365FOAdminToolkit/packages
+
+      # find and show content of any packages.config files
+      - name: Find and show packages.config
+        run: |
+          Get-ChildItem -Path . -Filter 'packages.config' -Recurse | ForEach-Object { 
+            Write-Host "***Showing content of: $($_.FullName)***"; 
+            Get-Content $_.FullName 
+          }
+
+      # find and show content of any nuget.config files
+      - name: Find and show NuGet.config files
+        run: |
+          Get-ChildItem -Path . -Filter 'nuget.config' -Recurse | ForEach-Object { 
+            Write-Host "***Showing content of: $($_.FullName)***"; 
+            Get-Content $_.FullName 
+          }
+
       - name: Restore NuGet packages
         run: nuget restore Project/D365FOAdminToolkit/D365FOAdminToolkitNET/D365FOAdminToolkitNET.csproj -PackagesDirectory Project/D365FOAdminToolkit/packages
 

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -80,20 +80,20 @@ jobs:
           version: ${{ matrix.version }}
           packagesDirectory: '${{ github.workspace }}/Project/D365FOAdminToolkit/packages'
 
-      - name: List FSC packages
+      - name: List FSC packages (debug)
         if: runner.debug == '1'
         run: ls -R Project/D365FOAdminToolkit/packages
 
-      # find and show content of any packages.config files
-      - name: Find and show packages.config
+      - name: Find and show packages.config (debug)
+        if: runner.debug == '1'
         run: |
           Get-ChildItem -Path / -Filter 'packages.config' -Recurse | ForEach-Object { 
             Write-Host "***Showing content of: $($_.FullName)***"; 
             Get-Content $_.FullName 
           }
 
-      # find and show content of any nuget.config files
-      - name: Find and show NuGet.config files
+      - name: Find and show NuGet.config files (debug)
+        if: runner.debug == '1'
         run: |
           Get-ChildItem -Path / -Filter 'nuget.config' -Recurse | ForEach-Object { 
             Write-Host "***Showing content of: $($_.FullName)***"; 
@@ -103,7 +103,8 @@ jobs:
       - name: Restore NuGet packages
         run: nuget restore Project/D365FOAdminToolkit/D365FOAdminToolkitNET/D365FOAdminToolkitNET.csproj -PackagesDirectory Project/D365FOAdminToolkit/packages
 
-      - name: List packages
+      - name: List packages (debug)
+        if: runner.debug == '1'
         run: ls -R Project/D365FOAdminToolkit/packages
 
       - name: Build

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -1,5 +1,11 @@
 on:
   pull_request:
+  # every day at half past 3 am
+  schedule:
+    - cron: '30 3 * * *'
+  push:
+    branches:
+      - dev
   workflow_dispatch:
    inputs:
       version:

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -83,6 +83,9 @@ jobs:
       - name: Restore NuGet packages
         run: nuget restore Project/D365FOAdminToolkit/D365FOAdminToolkitNET/D365FOAdminToolkitNET.csproj -PackagesDirectory Project/D365FOAdminToolkit/packages
 
+      - name: List packages
+        run: ls -R Project/D365FOAdminToolkit/packages
+
       - name: Build
         run: msbuild Project/D365FOAdminToolkit/D365FOAdminToolkitNET/D365FOAdminToolkitNET.csproj /p:VersionSuffix=pr /p:Configuration=Release /p:Version=0.0.0.${{ github.event.pull_request.number }} /p:InformationalVersion=0.0.0.${{ github.event.pull_request.number }}-pr -fileLogger
 

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -81,12 +81,13 @@ jobs:
           packagesDirectory: '${{ github.workspace }}/Project/D365FOAdminToolkit/packages'
 
       - name: List FSC packages
+        if: runner.debug == '1'
         run: ls -R Project/D365FOAdminToolkit/packages
 
       # find and show content of any packages.config files
       - name: Find and show packages.config
         run: |
-          Get-ChildItem -Path . -Filter 'packages.config' -Recurse | ForEach-Object { 
+          Get-ChildItem -Path / -Filter 'packages.config' -Recurse | ForEach-Object { 
             Write-Host "***Showing content of: $($_.FullName)***"; 
             Get-Content $_.FullName 
           }
@@ -94,7 +95,7 @@ jobs:
       # find and show content of any nuget.config files
       - name: Find and show NuGet.config files
         run: |
-          Get-ChildItem -Path . -Filter 'nuget.config' -Recurse | ForEach-Object { 
+          Get-ChildItem -Path / -Filter 'nuget.config' -Recurse | ForEach-Object { 
             Write-Host "***Showing content of: $($_.FullName)***"; 
             Get-Content $_.FullName 
           }

--- a/Project/D365FOAdminToolkit/D365FOAdminToolkitNET/D365FOAdminToolkitNET.csproj
+++ b/Project/D365FOAdminToolkit/D365FOAdminToolkitNET/D365FOAdminToolkitNET.csproj
@@ -225,7 +225,7 @@
     <None Include="packages.config" />
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <PropertyGroup>
+  <PropertyGroup Condition="Exists('$(ServiceDrive)\AosService\PackagesLocalDirectory\D365FOAdminToolkit\bin')">
     <PostBuildEvent>xcopy /y /c "$(TargetDir)*.dll" "$(ServiceDrive)\AosService\PackagesLocalDirectory\D365FOAdminToolkit\bin"</PostBuildEvent>
   </PropertyGroup>
 </Project>

--- a/Project/D365FOAdminToolkit/D365FOAdminToolkitNET/D365FOAdminToolkitNET.csproj
+++ b/Project/D365FOAdminToolkit/D365FOAdminToolkitNET/D365FOAdminToolkitNET.csproj
@@ -225,7 +225,11 @@
     <None Include="packages.config" />
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <PropertyGroup Condition="Exists('$(ServiceDrive)\AosService\PackagesLocalDirectory\D365FOAdminToolkit\bin')">
-    <PostBuildEvent>xcopy /y /c "$(TargetDir)*.dll" "$(ServiceDrive)\AosService\PackagesLocalDirectory\D365FOAdminToolkit\bin"</PostBuildEvent>
+  <PropertyGroup>
+    <PostBuildEvent>
+      if exist "$(ServiceDrive)\AosService\PackagesLocalDirectory\D365FOAdminToolkit\bin" (
+        xcopy /y /c "$(TargetDir)*.dll" "$(ServiceDrive)\AosService\PackagesLocalDirectory\D365FOAdminToolkit\bin"
+      )
+    </PostBuildEvent>
   </PropertyGroup>
 </Project>

--- a/Project/D365FOAdminToolkit/D365FOAdminToolkitNET/D365FOAdminToolkitNET.csproj
+++ b/Project/D365FOAdminToolkit/D365FOAdminToolkitNET/D365FOAdminToolkitNET.csproj
@@ -33,7 +33,11 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup Condition="Exists('..\packages')">
+    <Reference Include="..\packages\Microsoft.Dynamics.AX.Platform.CompilerPackage.*\Microsoft.Dynamics.AX.Metadata.dll" />
     <Reference Include="..\packages\Microsoft.Dynamics.AX.Platform.CompilerPackage.*\Microsoft.Dynamics.AX.Metadata.Core.dll" />
+    <Reference Include="..\packages\Microsoft.Dynamics.AX.Platform.CompilerPackage.*\Microsoft.Dynamics.AX.Metadata.Modeling.dll" />
+    <Reference Include="..\packages\Microsoft.Dynamics.AX.Platform.CompilerPackage.*\Microsoft.Dynamics.AX.Xpp.AxShared.dll" />
+    <Reference Include="..\packages\Microsoft.Dynamics.AX.Platform.CompilerPackage.*\Microsoft.Dynamics.AX.Xpp.Support.dll" />
     <Reference Include="..\packages\Microsoft.Dynamics.AX.Platform.DevALM.BuildXpp.*\ref\net40\ApplicationFoundation\bin\Microsoft.Dynamics.AX.Security.Management.dll" />
   </ItemGroup>
   <ItemGroup>

--- a/Project/D365FOAdminToolkit/D365FOAdminToolkitNET/D365FOAdminToolkitNET.csproj
+++ b/Project/D365FOAdminToolkit/D365FOAdminToolkitNET/D365FOAdminToolkitNET.csproj
@@ -55,23 +55,23 @@
     </Reference>
     <Reference Include="Microsoft.Dynamics.AX.Metadata, Version=7.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\..\..\AOSService\PackagesLocalDirectory\bin\Microsoft.Dynamics.AX.Metadata.dll</HintPath>
+      <HintPath>$(ServiceDrive)\AOSService\PackagesLocalDirectory\bin\Microsoft.Dynamics.AX.Metadata.dll</HintPath>
     </Reference>
     <Reference Include="Microsoft.Dynamics.AX.Metadata.Core">
-      <HintPath>..\..\..\..\..\AosService\WebRoot\bin\Microsoft.Dynamics.AX.Metadata.Core.dll</HintPath>
+      <HintPath>$(ServiceDrive)\AOSService\WebRoot\bin\Microsoft.Dynamics.AX.Metadata.Core.dll</HintPath>
     </Reference>
     <Reference Include="Microsoft.Dynamics.AX.Metadata.Modeling">
-      <HintPath>..\..\..\..\AOSService\PackagesLocalDirectory\bin\Microsoft.Dynamics.AX.Metadata.Modeling.dll</HintPath>
+      <HintPath>$(ServiceDrive)\AOSService\PackagesLocalDirectory\bin\Microsoft.Dynamics.AX.Metadata.Modeling.dll</HintPath>
     </Reference>
     <Reference Include="Microsoft.Dynamics.AX.Security.Management">
-      <HintPath>..\..\..\..\..\AosService\WebRoot\bin\Microsoft.Dynamics.AX.Security.Management.dll</HintPath>
+      <HintPath>$(ServiceDrive)\AOSService\WebRoot\bin\Microsoft.Dynamics.AX.Security.Management.dll</HintPath>
     </Reference>
     <Reference Include="Microsoft.Dynamics.AX.Xpp.AxShared, Version=7.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\..\..\AOSService\PackagesLocalDirectory\bin\Microsoft.Dynamics.AX.Xpp.AxShared.dll</HintPath>
+      <HintPath>$(ServiceDrive)\AOSService\PackagesLocalDirectory\bin\Microsoft.Dynamics.AX.Xpp.AxShared.dll</HintPath>
     </Reference>
     <Reference Include="Microsoft.Dynamics.AX.Xpp.Support">
-      <HintPath>..\..\..\..\AOSService\PackagesLocalDirectory\bin\Microsoft.Dynamics.AX.Xpp.Support.dll</HintPath>
+      <HintPath>$(ServiceDrive)\AOSService\PackagesLocalDirectory\bin\Microsoft.Dynamics.AX.Xpp.Support.dll</HintPath>
     </Reference>
     <Reference Include="Microsoft.Graph, Version=5.41.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\packages\Microsoft.Graph.5.41.0\lib\netstandard2.0\Microsoft.Graph.dll</HintPath>

--- a/Project/D365FOAdminToolkit/Get-DynamicsPackages.ps1
+++ b/Project/D365FOAdminToolkit/Get-DynamicsPackages.ps1
@@ -13,7 +13,11 @@ param (
 )
 
 
+Copy-Item -Path "$PackagesLocalDirectoryPath\bin\Microsoft.Dynamics.AX.Metadata.dll" -Destination "$RepoBasePath\Project\D365FOAdminToolkit\packages\DynamicsTemp"
 Copy-Item -Path "$PackagesLocalDirectoryPath\bin\Microsoft.Dynamics.AX.Metadata.Core.dll" -Destination "$RepoBasePath\Project\D365FOAdminToolkit\packages\DynamicsTemp"
+Copy-Item -Path "$PackagesLocalDirectoryPath\bin\Microsoft.Dynamics.AX.Metadata.Modeling.dll" -Destination "$RepoBasePath\Project\D365FOAdminToolkit\packages\DynamicsTemp"
+Copy-Item -Path "$PackagesLocalDirectoryPath\bin\Microsoft.Dynamics.AX.Xpp.AxShared.dll" -Destination "$RepoBasePath\Project\D365FOAdminToolkit\packages\DynamicsTemp"
+Copy-Item -Path "$PackagesLocalDirectoryPath\bin\Microsoft.Dynamics.AX.Xpp.Support.dll" -Destination "$RepoBasePath\Project\D365FOAdminToolkit\packages\DynamicsTemp"
 Copy-Item -Path "$PackagesLocalDirectoryPath\bin\Microsoft.Dynamics.AX.Security.Management.dll" -Destination "$RepoBasePath\Project\D365FOAdminToolkit\packages\DynamicsTemp"
 
 ###########################################

